Cython/Compiler/UtilityCode.py

```diff
@@ -265,6 +271,69 @@ def filter_inherited_directives(current_directives):
         return utility_code_directives
 
 
+class TemplatedFileSourceDescriptor(FileSourceDescriptor):
+
+    def __init__(self, filename, path_description, context):
+        super().__init__(filename, path_description)
+        self._context = context
+
+    def get_file_object(self, encoding=None, error_handling=None):
+        with super().get_file_object(encoding, error_handling) as f:
+            data = f.read()
+            ret = Code.sub_tempita(data, self._context, self.filename)
+            stream = io.TextIOWrapper(io.BytesIO(ret.encode(f.encoding)), encoding=f.encoding, errors=error_handling)
+        return stream
```

Comment:
This looks like it mixes two pieces of functionality into one. It's using the file *content* as template, not the file *descriptor*. Might just be bad class naming, but it feels wrong to see this combined.